File: Agora-RTM-Tutorial-iOS-Objective-C/Agora-RTM-Tutorial/PeerViewController.m
//
//  PeerViewController.m
//  Agora-Rtm-Tutorial
//
//  Created by CavanSu on 2019/2/19.
//  Copyright © 2019 Agora. All rights reserved.
//

#import "PeerViewController.h"
#import "ChatViewController.h"
#import "AgoraRtm.h"

@interface PeerViewController () <AgoraRtmDelegate>
@property (weak, nonatomic) IBOutlet UITextField *peerTextField;
@end

@implementation PeerViewController

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [AgoraRtm updateDelegate:self];
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    [self.view endEditing:YES];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    NSString *peer = (NSString *)sender;
    
    if ([segue.identifier isEqualToString:@"peerToChat"]) {
        ChatMode mode;
        mode.type = ChatTypePeer;
        mode.name = peer;
        
        ChatViewController *chatVC = (ChatViewController *)segue.destinationViewController;
        chatVC.mode = mode;
    }
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    [self.view endEditing:true];
}

- (IBAction)doChatPressed:(UIButton *)sender {
    NSString *peer = self.peerTextField.text;
    
    if (!peer.length) {
        return;
    }
    
    [self performSegueWithIdentifier:@"peerToChat" sender:peer];
}

- (void)rtmKit:(AgoraRtmKit *)kit connectionStateChanged:(AgoraRtmConnectionState)state reason:(AgoraRtmConnectionChangeReason)reason {
    NSString *message = [NSString stringWithFormat:@"connection state changed: %ld", state];
    __weak PeerViewController *weakSelf = self;
    [self showAlert:message handle:^(UIAlertAction * action) {
        if (reason == AgoraRtmConnectionChangeReasonRemoteLogin) {
            [weakSelf.navigationController popToRootViewControllerAnimated:YES];
        }
    }];
}

@end

File: Tutorial-iOS&Mac/Agora-Rtm-Tutorial/PeerViewController.swift
//
//  PeerViewController.swift
//  Agora-Rtm-Tutorial
//
//  Created by CavanSu on 2019/1/17.
//  Copyright © 2019 Agora.IO. All rights reserved.
//

import UIKit

class PeerViewController: UIViewController {
    @IBOutlet weak var peerTextField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    @IBAction func doChatPressed(_ sender: UIButton) {
        guard let peer = peerTextField.text, peer.count > 0 else {
            return
        }
        performSegue(withIdentifier: "peerToChat", sender: peer)
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        guard let identifier = segue.identifier, let peer = sender as? String else {
            return
        }
        
        switch identifier {
        case "peerToChat":
            let chatVC = segue.destination as! ChatViewController
            chatVC.type = .peer(peer)
        default:
            break
        }
    }
}
